test/powershell/Modules/Microsoft.PowerShell.Utility/Get-FileHash.Tests.ps1

```diff
@@ -77,4 +77,17 @@ Describe "Get-FileHash" -Tags "CI" {
             $result.Path | Should -Be $testDocument
         }
     }
+    
+    Context "Close file before writing hash details to pipeline" {
+        It "Should not hold the file open unnecessarily" {
+            # Make a copy of the test document first,
+            # because this test will rename it.
+            $testDocumentCopy = "${testDocument}-copy"
+            Copy-Item -Path $testdocument -Destination $testDocumentCopy
+            
+            {Get-FileHash -Path $testDocumentCopy | Rename-Item -NewName {$_.Hash} -ErrorAction Stop} | Should -Not -Throw
```

Comment:
Positive check is better because we don't expect throw but a result:
```suggestion
            { Get-FileHash -Path $testDocumentCopy | Rename-Item -NewName { $_.Hash } -PassThru } | Should -Exist
```

Reply:
Good idea! I have updated it with a positive test using `Test-Path` to look for the renamed file, and using `-ErrorAction SilentlyContinue` so if the rename fails, the tests will behave predictably.